torch/_subclasses/functional_tensor.py

```diff
@@ -17,6 +17,27 @@
 not_implemented_log = torch._logging.getArtifactLogger(__name__, "not_implemented")
 
 
+# NOTE Some special handling for tensor conversion during export is needed.
+# Normally, when tracing through the model with tensor.to(), the maybe-aliasing
+# relationship between input and output tensors will be baked into the graph.
+# For example, if we got a tensor with device cpu and call tensor.to("cpu"),
+# it will become a no-op in the graph. For a whole graph capture, this is not
+# sound so we need to do something different. Instead, in export we will try to
```

Comment:
Wait, this banning of mutations only happens in export? But this logic is used from other places right? Which would make them unsound?

Reply:
> But this logic is used from other places right?

This logic is only used in export right now. We made a check in functionalization to only trigger this under export.

Reply:
Ok!